src/components/PopupWithForm.js

```diff
@@ -1,36 +1,46 @@
-import PopUp from "./PopUp.js";
+import PopUp from "./Popup.js";
 
 export default class PopUpWithForm extends PopUp {
   constructor(popupSelector, handleFormSubmit) {
     super({ popupSelector });
     this._popupForm = this._popupElement.querySelector(".modal__form");
     this._handleFormSubmit = handleFormSubmit;
     this._inputEls = Array.from(
-      this._popupForm.querySelectorAll(".modal__input")
-    );
+      this._popupForm.querySelectorAll(".modal__input"));
+    this._popupButton = this._popupForm.querySelector(".modal__button");
   }
+  
 
   close() {
     this._popupForm.reset();
     super.close();
   }
 
   _getInputValues() {
-    const formValues = [];
+    const formValues = {};
     this._inputEls.forEach((element) => {
       formValues[element.name] = element.value;
     });
 
     return formValues;
   }
 
+  submitLoading(isLoading) {
+    if (isLoading) {
+      this._popupButton.textContent = "Saving...";
+    } else {
+      this._popupButton.textContent = "Save";
```

Comment:
**Could be improved**

It’s better to make method `renderLoading` inside class `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```